### ISSUE TITLE
Failure on julia 0.7: Coverage 0.7.0 incompatible with HTTP 0.8.0?

### ISSUE THREAD

**Issue by martinholters** [JuliaCI/Coverage.jl#202]
From a recent Travis job on Julia 0.7:
```
Submitting data to Coveralls...
ERROR: MethodError: no method matching setheader(::Array{Pair{String,String},1}, ::Pair{String,String})
Closest candidates are:
  setheader(!Matched::HTTP.Messages.Message, ::Any) at /home/travis/.julia/packages/HTTP/GN0Te/src/Messages.jl:345
  setheader(!Matched::Array{Pair{SubString{String},SubString{String}},1}, ::Pair) at /home/travis/.julia/packages/HTTP/GN0Te/src/Messages.jl:347
  setheader(!Matched::HTTP.Streams.Stream, ::Any...) at /home/travis/.julia/packages/HTTP/GN0Te/src/Streams.jl:56
Stacktrace:
 [1] #post#4(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::String, ::Array{Pair{String,String},1}, ::HTTP.Form) at /home/travis/.julia/packages/HTTP/GN0Te/src/multipart.jl:131
 [2] post(::String, ::Array{Pair{String,String},1}, ::HTTP.Form) at /home/travis/.julia/packages/HTTP/GN0Te/src/multipart.jl:131
 [3] #submit#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Array{FileCoverage,1}) at /home/travis/.julia/packages/Coverage/yjStK/src/coveralls.jl:107
 [4] submit(::Array{FileCoverage,1}) at /home/travis/.julia/packages/Coverage/yjStK/src/coveralls.jl:60
 [5] top-level scope at none:0
```
Pinning HTTP to 0.7.1 (instead of the current 0.8.0) seems to fix this. I've noticed that Julia 1.0 and later use Coverage 0.8.1, while Julia 0.7 uses Coverage 0.7.0. So might it be that Coverage 0.7.0 is incompatible with HTTP 0.8.0. Is that the case? Then we should update the compat info in METADATA.

(Note: At first I thought this might be an issue with HTTP on Julia 0.7 (ref https://github.com/JuliaWeb/HTTP.jl/issues/388), but I've become doubtful of that.

**Comment by martinholters** [comment]
Given #197, I think we should upper-bound HTTP to be < 0.8.0 for Coverage up to (including) 0.8.0. (The fix made it into 0.8.1). Will open at PR to METADATA.

**Comment by martinholters** [comment]
Resolved by https://github.com/JuliaLang/METADATA.jl/pull/21711
